authui/src/colorscheme.ts

```diff
@@ -1,43 +1,31 @@
-import { Controller } from "@hotwired/stimulus";
-
-export class ColorSchemeController extends Controller {
-  static values = {
-    darkThemeEnabled: Boolean,
-  };
-
-  queryResult = window.matchMedia("(prefers-color-scheme: dark)");
-
-  declare darkThemeEnabledValue: Boolean;
-
-  onChange = () => {
-    if (!this.darkThemeEnabledValue) {
-      return;
-    }
-
-    let explicitColorScheme = "";
-    const metaElement = document.querySelector("meta[name=x-color-scheme]");
-    if (metaElement instanceof HTMLMetaElement) {
-      explicitColorScheme = metaElement.content;
-    }
+function onChange() {
+  const htmlElement = document.getElementsByTagName("html")[0];
+  const darkThemeEnabledValue = htmlElement.getAttribute(
+    "data-dark-theme-enabled"
+  );
+
+  if (!darkThemeEnabledValue) {
+    return;
+  }
 
-    const implicitColorScheme = this.queryResult.matches ? "dark" : "light";
+  let explicitColorScheme = "";
+  const metaElement = document.querySelector("meta[name=x-color-scheme]");
+  if (metaElement instanceof HTMLMetaElement) {
+    explicitColorScheme = metaElement.content;
+  }
 
-    const colorScheme =
-      explicitColorScheme !== "" ? explicitColorScheme : implicitColorScheme;
+  const implicitColorScheme = queryResult.matches ? "dark" : "light";
 
-    if (colorScheme === "dark") {
-      this.element.classList.add("dark");
-    } else {
-      this.element.classList.remove("dark");
-    }
-  };
+  const colorScheme =
+    explicitColorScheme !== "" ? explicitColorScheme : implicitColorScheme;
 
-  connect() {
-    this.queryResult.addEventListener("change", this.onChange);
-    this.onChange();
-  }
-
-  disconnect() {
-    this.queryResult.removeEventListener("change", this.onChange);
+  if (colorScheme === "dark") {
+    htmlElement.classList.add("dark");
+  } else {
+    htmlElement.classList.remove("dark");
   }
 }
+
+const queryResult = window.matchMedia("(prefers-color-scheme: dark)");
+queryResult.addEventListener("change", onChange);
+onChange();
```

Comment:
So now `colorscheme.ts` is a classic script. It is being executed in the global scope. After evaluating it, we now have polluted the global scope with two global variables `onChange` and `queryResult`. Classic script usually takes the form of [IIFE](https://developer.mozilla.org/en-US/docs/Glossary/IIFE). Convert this classic script into IIFE to avoid polluting the global scope.